### ISSUE TITLE
815-01-Win64 does not open because of libcurl

### ISSUE THREAD

**Issue by serrufs** [simulationcraft/simc#4778]
SimcSetup-815-01-Win64 does not open in win 8.1 x64
installed
Microsoft Visual C++ 2017 Redistributable (x64) - 14.20.27404
Microsoft Visual C++ 2017 Redistributable (x86) - 14.20.27404

opens and closes immediately...

**Comment by navv1234** [comment]
Not really sure why it would behave like that. The installer works fine on the few win10 boxes, and a win7 box I tested it on.

No chance this could be some antivirus being overzealous?

**Comment by serrufs** [comment]
I disable antivirus - it does not help
same with sims-810-02-win64*****.7z - thought then, is because they do not release
SimcSetup-810-01-Win64.exe = works is always
and new versions do not work ((

<a href="https://jpegshare.net/1b/32/1b328dcbc5cfb3fa4ed995f219cca26d.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/1b/32/1b328dcbc5cfb3fa4ed995f219cca26d.jpg"/><a>

**Comment by navv1234** [comment]
The 7z files are compressed archives and should work fine. You need to
download and install the application to unpack them from
https://www.7-zip.org

Then unpack the archive to a directory of your choosing and run
simulationcraft.exe from there.

**Comment by serrufs** [comment]
I understand)) I write that the problem is reproduced starting from this version 810-02...
Version -810-01 works fine for me

**Comment by navv1234** [comment]
Right. The only thing changed between the 810 01 release and 815 01 that
affects this is probably the installer sdk version. I can try to revert
that to the old one at some point.

**Comment by serrufs** [comment]
<a href="https://jpegshare.net/64/ac/64ac600b63ead79828209af903376b10.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/64/ac/64ac600b63ead79828209af903376b10.jpg"/><a>
I put VCR from your package - it does not help ((

**Comment by navv1234** [comment]
If it is the installer program that does not work then updating vcr
probably won't work. If it is simulationcraft.exe then that is a whole
another issue.

**Comment by serrufs** [comment]
the program 815 01 is installing - as far as this happens correctly, I can not judge, so I think the problem is in simulationcraft.exe

I installed back the new vcr
<a href="https://jpegshare.net/0f/d5/0fd5d67e4a834a22524fc856677e4948.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/0f/d5/0fd5d67e4a834a22524fc856677e4948.jpg"/><a>

I deleted the following folders
<a href="https://jpegshare.net/76/67/76676a3ad5e0df57a1d011581eb4945e.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/76/67/76676a3ad5e0df57a1d011581eb4945e.jpg"/><a>
that didn't help either ((

I recorded the installation video:
https://yadi.sk/i/HQ8k7TZu8F7trQ
and launch video
https://yadi.sk/i/dI460CwtrtVTHg

maybe it will help to understand what the problem is, if you need any additional data, write, I will provide

**Comment by navv1234** [comment]
You could try https://github.com/simulationcraft/simc/wiki/CommonIssues point 3. It could be that the version of chromium used by qt is somehow allergic to your gpu drivers. It has been known to happen.

**Comment by serrufs** [comment]
I added a variable environment - but it didn't help ((
the program still closes immediately ...
here are my gpu drivers
<a href="https://jpegshare.net/c0/54/c0546bc97e47a3dcb9e1305bea548ab3.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/c0/54/c0546bc97e47a3dcb9e1305bea548ab3.jpg"/><a>

**Comment by navv1234** [comment]
Does http://downloads.simulationcraft.org/simc-815-02-win64-801a32516b-qt597.7z work better?

**Comment by serrufs** [comment]
All, too, most 810-01 works well, and 815-02 from file sims-815-02-win64-801a32516b-qt597.7z does not start unfortunately (((((((((((

**Comment by serrufs** [comment]
This is my PC with Win 8.1 x64 (Home):
<a href="https://jpegshare.net/e6/ed/e6eda0b6b1e169d2d7ab9292e89d1f76.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/e6/ed/e6eda0b6b1e169d2d7ab9292e89d1f76.jpg"/><a>
<a href="https://jpegshare.net/" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/images/14/33/143354b6a9b8c281caaf29acd5079443.png"/></a>

I also tried running this file on another device( laptop) with Win 8.1 x64 RU:
<a href="https://jpegshare.net/" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/images/1c/59/1c592170f2c69d38c004b683588015ea.jpg"/></a> +GTX860M
It also does not start 815-02 (((

**Comment by navv1234** [comment]
How about this one http://downloads.simulationcraft.org/simc-815-02-win64-801a32516b-notr.7z ?

**Comment by serrufs** [comment]
still this version also does not start ((

**Comment by navv1234** [comment]
Aand how about this one: http://downloads.simulationcraft.org/simc-815-02-win64-801a32516b-nonetwork.7z

**Comment by serrufs** [comment]
this one started, just before opening it gave a warning
<a href="https://jpegshare.net/" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/images/18/6a/186a21accc5fc7b6db670eadbae41834.jpg"/></a>
thank you now it works
<a href="https://jpegshare.net/09/7d/097d694f570e02a5614e155423be424e.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/09/7d/097d694f570e02a5614e155423be424e.jpg"/><a>
What was the problem?))

**Comment by navv1234** [comment]
Curl, the library we use to download characters (import them) from armory now that blizzard has transitioned to a more complicated authentication system on their end.

That release is not going to be able to import them from the armory, unfortunately. What you could do is download https://github.com/lucasg/Dependencies this program,  open the "DependenciesGUI" and load Simulationcraft.exe from one of the packages that do not work. On the left side you'll see a bunch of DLL files in a tree-like structure, one of them is "libcurl.dll". If you expand that one, is there something missing from the DLLs "libcurl.dll" itself uses?

**Comment by serrufs** [comment]
You don't understand, it works for me. This window appears only once and as I understand it resets SimulationCraft.ini

**Comment by navv1234** [comment]
I understand it works for you, but this is not a release we will be building because you cannot import anything with it from Blizzard Armory. So I would need to understand why libcurl causes this for you to properly fix it in the real releases.

**Comment by serrufs** [comment]
I use https://www.curseforge.com/wow/addons/simulationcraft so I don't need import with with it from Blizzard Armory
but I'm ready to help you, write what you need to do

**Comment by navv1234** [comment]
"What you could do is download https://github.com/lucasg/Dependencies this program, open the "DependenciesGUI" and load Simulationcraft.exe from one of the packages that do not work. On the left side you'll see a bunch of DLL files in a tree-like structure, one of them is "libcurl.dll". If you expand that one, is there something missing from the DLLs "libcurl.dll" itself uses?"

**Comment by serrufs** [comment]
And there is no search for these problem "?"branches? ))) And that I open 30 minutes directories, subdirectories, subdirectories of subdirectories, subdirectories of subdirectories and deeper and deeper, there is definitely not a closed infinite loop? )))
<a href="https://jpegshare.net/20/0e/200e89c7db6891646cd37bfb2c354e25.png.html" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/thumbs/20/0e/200e89c7db6891646cd37bfb2c354e25.jpg"/><a>
so maybe I'm not doing something now will try again skip this branch ..

**Comment by serrufs** [comment]
directory C:\Users\Администратор\Desktop\simc-815-01-win64\libcurl.dll

<a href="https://ibb.co/9cK6zn0"><img src="https://i.ibb.co/9cK6zn0/0001.png" alt="0001" border="0"></a> <a href="https://ibb.co/ZGJh85N"><img src="https://i.ibb.co/ZGJh85N/0002.png" alt="0002" border="0"></a> <a href="https://ibb.co/JBj81dj"><img src="https://i.ibb.co/JBj81dj/0003.png" alt="0003" border="0"></a> <a href="https://ibb.co/h9Pf37n"><img src="https://i.ibb.co/h9Pf37n/0004.png" alt="0004" border="0"></a> <a href="https://ibb.co/FgLhKnm"><img src="https://i.ibb.co/FgLhKnm/0005.png" alt="0005" border="0"></a> <a href="https://ibb.co/rphHT94"><img src="https://i.ibb.co/rphHT94/0006.png" alt="0006" border="0"></a>

**Comment by navv1234** [comment]
That all looks as it should, apparently. I guess I'll try to find a win81 machine from somewhere.

**Comment by serrufs** [comment]
ok

**Comment by navv1234** [comment]
Alright, well a fresh install of windows 8.1 pro on a virtual machine can launch the latest nightly built on the site (http://downloads.simulationcraft.org/simc-815-02-win64-5556fcc9cb.7z), so i'm not really sure what the problem could be.

**Comment by serrufs** [comment]
We have version 8.1 core (Home).
And this version http://downloads.simulationcraft.org/simc-815-02-win64-5556fcc9cb.7z does not run on a PC not on a laptop((((
Have you installed all the updates? We have video from NVIDIA with drivers 419.35 on both computers

I don't know what else could be ((

**Comment by navv1234** [comment]
It's a virtual machine so not much to update, honestly. And it's not the GPU drivers, the problem is "libcurl.dll" almost certainly, I just don't know why.

**Comment by navv1234** [comment]
Can you try if you can start http://downloads.simulationcraft.org/simc-815-02-win64-215d7b7dbe-openssl.7z this properly?

**Comment by serrufs** [comment]
Does not open this version and with 28000@license-1.cae.wisc.edu and without this variable

until works only 810 01 and http://downloads.simulationcraft.org/simc-815-02-win64-801a32516b-nonetwork.7z without this variable

**Comment by serrufs** [comment]
I downloaded the new version simc-815-02-win64-431464566b.7z , now it shows me error that the system is missing libssl-1_1-x64.dll
<a href="https://jpegshare.net/" target="_blank" title="JpegShare.net - бесплатный хостинг картинок"><img src="https://jpegshare.net/images/f7/d7/f7d7bbde6093960736460a3a13d2c907.jpg"/></a>

**Comment by navv1234** [comment]
Yeah, was built with incorrect libcurl linkage. It is now fixed in the latest nightly. Also, there is a "nonetwork" version of the nightly simc now, that does not link to libcurl.dll. It should work in your case.

**Comment by serrufs** [comment]
OK, thanks

**Comment by metalgrid** [comment]
Same issue, but on:
```
OS Name	Microsoft Windows 10 Home
Version	10.0.17763 Build 17763
```
DependenciesGui doesn't show anything missing, can provide screenshots if needed.
The nonetwork version works smoothly.
I tried both win32 and win64 versions - same symptoms: opening SimulationCraft.exe produces a loading cursor for a second and then nothing. Invoking simc.exe inside a terminal provides no output.
I tried replacing libcurl.dll with the latest binary version from the official website but it didn't change behaviour (other than requiring libcrypto-x64.dll and libssl-x64.dll which I also got from the same location).

No antivirus software running, other than windows defender, but disabling that doesn't affect the behaviour as well. Adding libcurl.dll manually to an exception list also does nothing.

Let me know if I can do anything else to diagnose, as I'm very interested why this is acting like this.
